Python/errors.c

```diff
@@ -301,6 +301,23 @@ PyErr_SetString(PyObject *exception, const char *string)
     _PyErr_SetString(tstate, exception, string);
 }
 
+void
+_PyErr_SetLocaleStringTstate(PyThreadState *tstate, PyObject *exception,
+                             const char *string)
+{
+    PyObject *value = PyUnicode_DecodeLocale(string, "strict");
```

Comment:
I think this should be `surrogateescape`, not `strict`:
```suggestion
    PyObject *value = PyUnicode_DecodeLocale(string, "surrogateescape");
```

Reply:
Both are acceptable and I think it's preferrable to use a strict decoding since a message that is not acceptable by the current locale feels wrong. In addition, using surrogateescape means that for re-encoding the message you need to also pass the "surrogateescape" handler.

Reply:
Ok, looking around, other usages in the context of error rendering use the "surrogateescape" handler, so I've decided to use it as well.